src/editorconfig.rs

```diff
@@ -0,0 +1,294 @@
+#[cfg(feature = "editorconfig")]
+use {
+    crate::{CallParenType, CollapseSimpleStatement, Config, IndentType, LineEndings, QuoteStyle},
+    ec4rs::{
+        properties_of,
+        property::{EndOfLine, IndentSize, IndentStyle, MaxLineLen},
+        rawvalue::RawValue,
+        Properties,
+    },
+    std::path::Path,
+};
+
+#[cfg(feature = "editorconfig")]
+pub fn load_editorconfig(config: Config, directory: &Path) -> Config {
+    // HACK: used to detect filetype, this is not a real file path or glob
+    let path = directory.join("*.lua");
+
+    // Read the EditorConfig files that would apply to a file at the given path
+    let mut cfg = properties_of(path).unwrap();
+
+    cfg.use_fallbacks();
+
+    load_editorconfig_overrides(config, &cfg)
+}
+
+fn parse_string(raw: &RawValue) -> String {
+    raw.filter_unset().into_str().to_lowercase()
+}
+
+fn parse_integer(raw: &RawValue) -> Result<usize, std::num::ParseIntError> {
+    raw.filter_unset().into_str().parse::<usize>()
+}
+
+#[cfg(feature = "editorconfig")]
+fn load_editorconfig_overrides(config: Config, cfg: &Properties) -> Config {
+    let mut new_config = config;
+
+    // EditorConfig end_of_line
+    match parse_string(cfg.get_raw::<EndOfLine>()).as_str() {
```

Comment:
Is there a reason why in all of these that we use the raw version rather than the preprocessed properties and their corresponding enums https://docs.rs/ec4rs/latest/ec4rs/property/index.html

Reply:
I thought `cfg.get` would not lowercase the values (but it does), and this is now replaced with `cfg.iter` + `value.parse`, which does too.